nomad/structs/network.go

```diff
@@ -51,9 +56,9 @@ type NetworkIndex struct {
 // NewNetworkIndex is used to construct a new network index
 func NewNetworkIndex() *NetworkIndex {
 	return &NetworkIndex{
-		AvailAddresses: make(map[string][]NodeNetworkAddress),
-		AvailBandwidth: make(map[string]int),
+		HostNetworks:   make(map[string][]NodeNetworkAddress),
```

Comment:
At first I was thinking it would be safer to create the `TaskNetworks` and `GroupNetworks` slices here too, but I'm realizing by leaving them out we effectively enforce that `SetNode` has been called before use (on pain of panic). So :+1: here